src/core/qgsdataitem.cpp

```diff
@@ -1287,51 +1285,8 @@ void QgsZipItem::init()
 
   if ( sProviderNames.isEmpty() )
   {
-    // QStringList keys = QgsProviderRegistry::instance()->providerList();
-    // only use GDAL and OGR providers as we use the VSIFILE mechanism
-    QStringList keys;
-    // keys << "ogr" << "gdal";
-    keys << QStringLiteral( "gdal" ) << QStringLiteral( "ogr" );
-
-    for ( const auto &k : qgis::as_const( keys ) )
-    {
-      QgsDebugMsgLevel( "provider " + k, 3 );
-      // some providers hangs with empty uri (PostGIS) etc...
-      // -> using libraries directly
-      std::unique_ptr< QLibrary > library( QgsProviderRegistry::instance()->createProviderLibrary( k ) );
-      if ( library )
-      {
-        dataCapabilities_t *dataCapabilities = reinterpret_cast< dataCapabilities_t * >( cast_to_fptr( library->resolve( "dataCapabilities" ) ) );
-        if ( !dataCapabilities )
-        {
-          QgsDebugMsg( library->fileName() + " does not have dataCapabilities" );
-          continue;
-        }
-        if ( dataCapabilities() == QgsDataProvider::NoDataCapabilities )
-        {
-          QgsDebugMsg( library->fileName() + " has NoDataCapabilities" );
-          continue;
-        }
-        QgsDebugMsg( QString( "%1 dataCapabilities : %2" ).arg( library->fileName() ).arg( dataCapabilities() ) );
-
-        dataItem_t *dataItem = reinterpret_cast< dataItem_t * >( cast_to_fptr( library->resolve( "dataItem" ) ) );
-        if ( ! dataItem )
-        {
-          QgsDebugMsg( library->fileName() + " does not have dataItem" );
-          continue;
-        }
-
-        // mLibraries.append( library );
-        sDataItemPtr.append( dataItem );
-        sProviderNames.append( k );
-      }
-      else
-      {
-        //QgsDebugMsg ( "Cannot get provider " + k );
-      }
-    }
+    sProviderNames << QStringLiteral( "OGR" ) << QStringLiteral( "OGR" );
```

Comment:
Why adding OGR twice? Did you mean GDAL?

Reply:
Fixed.